api.py

```diff
@@ -469,6 +454,12 @@ def request_data(self, course_id=None, date=None, next_page_token=None):
             )
         )
 
+    def preprocess_records(self, records):
```

Comment:
@zkagin Does this approach look better to you? It seems a little less dry and only works for these specific endpoints, but since that is the intended use case I think it works fine.